src/screens/Register.jsx

```diff
@@ -21,21 +32,53 @@ const Register = () => {
   };
 
   const assignUserRole = async (email, values) => {
-    const account = await assignRole({ email, ...values });
-    console.log(account);
+    if (role === "candidate") {
+      const account = await assignCandidate({ email, ...values });
+      console.log(account);
+    } else {
+      const account = await assignCompany({ email, ...values });
+      console.log(account);
+    }
   };
 
-  if (isLoading) return <Loader />;
+  if (isLoading && role) return <Loader />;
```

Comment:
You should not check for the role here. If you set your role here after a user clicks on assign role, I think you would end up with a `<Loader>` component forever.